pandas/_libs/testing.pyx

```diff
@@ -159,12 +160,14 @@ cpdef assert_almost_equal(a, b,
             except AssertionError:
                 is_unequal = True
                 diff += 1
+                if not first_diff:
+                    first_diff = f"At index {i} diff: {a[i]} != {b[i]}"
```

Comment:
```suggestion
                    first_diff = f"At index {i}, first diff: {a[i]} != {b[i]}"
```

What do you think? Was thinking to clarify that this may not be the only diff during the comparison.